lib/idempotency.rb

```diff
@@ -21,4 +22,72 @@ module Idempotency
   def self.cache
     @cache ||= Cache.new(config:)
   end
+
+  def self.use_cache(request, request_identifiers, lock_duration: nil, &blk)
+    new.use_cache(request, request_identifiers, lock_duration:, &blk)
+  end
+
+  def use_cache(request, request_identifiers, lock_duration:) # rubocop:disable Metrics/AbcSize
+    return yield unless cache_request?(request)
+
+    cache = Idempotency.cache
```

Comment:
Could we pass `cache` and `config` via initializer so that we can mock easier in test?